src/html/document.zig

```diff
@@ -81,14 +84,17 @@ pub const HTMLDocument = struct {
         }
     }
 
-    // TODO: not implemented by libdom
-    pub fn get_cookie(_: *parser.DocumentHTML) ![]const u8 {
-        return error.NotImplemented;
+    pub fn get_cookie(_: *parser.DocumentHTML, arena: std.mem.Allocator, userctx: UserContext) ![]const u8 {
+        var buf: std.ArrayListUnmanaged(u8) = .{};
+        try userctx.cookie_jar.forRequest(&userctx.url.uri, buf.writer(arena), .{ .navigation = true });
+        return buf.items;
     }
 
-    // TODO: not implemented by libdom
-    pub fn set_cookie(_: *parser.DocumentHTML, _: []const u8) ![]const u8 {
-        return error.NotImplemented;
+    pub fn set_cookie(_: *parser.DocumentHTML, arena: std.mem.Allocator, userctx: UserContext, cookie_str: []const u8) ![]const u8 {
+        const c = try Cookie.parse(arena, &userctx.url.uri, cookie_str);
```

Comment:
This is a problem, the cookie has to outlive the arena (cookies live for the duration of the session, the arena is for the page). You can parse with `userctx.cookie_jar.allocator`  instead of `arena`. 

Not the first case where I think _most_ web apis need the page_arena, but a few places need something with a longer lifetime. In `jsuruntime`, there is no special parameter for Allocator (or Loop), there's simply a `session_state.arena`. We could add a `session_state.session_arena`  and/or a `session_state.allocator`

Reply:
Good catch :+1: 

Did you see other places in that case in web api implementations?

Reply:
Somewhat related, we see this problem with XHR, where the page_arena is disposed of, clearing out the XHR object, but there are still inflight IO events. It crashes the app. It happens if the page load is slow, and the CDP client disconnects due to a timeout. It's on my radar.

The only other thing which I think currently outlives the page is the storage shed, but I just looked at the code again, and it seems OK as it's using the Session's arena.